pkg/api/types.go

```diff
@@ -172,6 +172,9 @@ type Volume struct {
 	// This is optional for now. If not specified, the Volume is implied to be an EmptyDir.
 	// This implied behavior is deprecated and will be removed in a future version.
 	VolumeSource `json:",inline,omitempty"`
+	// SecretName is the name of a Secret in the same namespace as the pod containing this volume
+	// used for volume authentication
+	SecretName string `json:"secretName,omitempty"`
```

Comment:
No, this is a different one.  The end user will have to understand how to pair a secret with a volume.  How will they know?  Good docs?  That's a bit of a cop-out.  @pmorie has the use case for this.  Perhaps he has a good answer.

Ceph is our use case for a secret on a PV.  Putting the onus on the admin to understand how to use a SecretRef is reasonable with good supporting documentation for the plugin.